components/application-deployers/org.wso2.carbon.application.deployer.synapse/src/main/java/org/wso2/carbon/application/deployer/synapse/SynapseAppDeployer.java

```diff
@@ -134,6 +145,8 @@ public void deployArtifacts(CarbonApplication carbonApp, AxisConfiguration axisC
                 }
             }
         }
+        // Add the CApp artifacts details for specific tenant
+        cAppArtifactDataService.addCAppArtifactData(tenantId, cAppArtifacts);
```

Comment:
Keeping 1 map for each tenant & holding all CAPP artifacts inside the map. 
When ever the CAPP is uploaded, the artifacts will be append in the existing map.
The implementation is added in 'CAppDataHolder' class under 'addCAppArtifactData' method.